components/client/src/main/java/com/hotels/styx/client/connectionpool/StatsReportingConnectionPool.java

```diff
@@ -81,11 +81,16 @@ public void close() {
         removeMetrics();
     }
 
-    private void removeMetrics() {
-        MetricRegistry scopedRegistry = getMetricScope(connectionPool);
-        asList("busy-connections", "pending-connections", "available-connections", "ttfb",
-                "connection-attempts", "connection-failures", "connections-closed", "connections-terminated")
-                .forEach(scopedRegistry::deregister);
+    private static void registerMetrics(ConnectionPool hostConnectionPool, MetricRegistry scopedRegistry) {
+        ConnectionPool.Stats stats = hostConnectionPool.stats();
+        scopedRegistry.register("busy-connections", (Gauge<Integer>) stats::busyConnectionCount);
+        scopedRegistry.register("pending-connections", (Gauge<Integer>) stats::pendingConnectionCount);
+        scopedRegistry.register("available-connections", (Gauge<Integer>) stats::availableConnectionCount);
+        scopedRegistry.register("connection-attempts", (Gauge<Integer>) () -> (int) stats.connectionAttempts());
+        scopedRegistry.register("connection-failures", (Gauge<Integer>) () -> (int) stats.connectionFailures());
+        scopedRegistry.register("connections-closed", (Gauge<Integer>) () -> (int) stats.closedConnections());
+        scopedRegistry.register("connections-terminated", (Gauge<Integer>) () -> (int) stats.terminatedConnections());
+        scopedRegistry.register("in-establishment", (Gauge<Integer>) () -> (int) stats.connectionsInEstablishment());
```

Comment:
Should we keep the same terminology here? All the others have 'connections' in the name.

Reply:
I will change this later when fixing the pool error.